meshapp/src/profile/profile-name.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from "react";
+import { TextField, Box, ThemeProvider, Theme, Typography, Grid, createTheme } from "@mui/material";
+
+import EditIcon from "@mui/icons-material/Edit";
+import SaveIcon from "@mui/icons-material/Save";
+
+
+/**
+ * A React component that renders a text field with editing capabilities.
+ * Includes a character limit and an edit/save mode toggle.
+ *
+ * Used in the Profile page (src/profile/profile-page.tsx).
+ *
+ * @param props - Properties of the component
+ * @param {string} props.label - The label
+ * @param {string} props.placeholder - The placeholder text
+ * @param {string} props.text - The initial text content
+ * @param {number} props.charLimit - The max number of characters allowed
+ */
+const theme = createTheme({
+    palette: {
+      mode: "light",
+      primary: {
+        main: "#0b7d66",
+      },
+    },
+    typography: {
+      h1: {
+        fontFamily: "cocogoose",
+        fontWeight: "bold",
+        color: "#26383A",
+      },
+    },
+});
+
+
+
+const ProfileName = (props: {
+  label: string;
+  placeholder: string;
+  text: string;
+  charLimit: number;
+}) => {
+  const [text, setText] = useState(props.text);
+  const [editMode, setEditMode] = useState(false);
+  
+
+  // Enforce developer-defined character limit and empty string
+  const handleTextChange = (event: any) => {
+    if (event.target.value.length > props.charLimit) return;
+    if (event.target.value.trim() === "") return;
+    if (!/^[a-zA-Z\s]+$/.test(event.target.value)) return;
+    setText(event.target.value);
```

Comment:
For all three of these, there should be a little more than just a return. Ideally, you would tell the user what was wrong.

Reply:
Used Snackbar to tell user what's wrong.

testing comment